Classes/TextUtils.cs

```diff
@@ -24,19 +24,84 @@ namespace RadioDld
+        /// Will only remove dates within a set range eitherside of the programme date.
+        /// </summary>
+        /// <param name="name">Episode Name to be checked.</param>
+        /// <param name="stripDate">Date value to be striped from the Episode Name.</param>
+        /// <returns>Episode name with date content removed</returns>
         public static string StripDateFromName(string name, DateTime stripDate)
         {
-            // Use regex to remove a number of different date formats from episode titles.
-            // Will only remove dates with the same month & year as the programme itself, but any day of the month
-            // as there is sometimes a mismatch of a day or two between the date in a title and the publish date.
-            Regex matchStripDate = new Regex("\\A(" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "/" + stripDate.ToString("MM", CultureInfo.InvariantCulture) + "/\\d{2} ?-? )?(?<name>.*?)( ?:? (\\d{2}/" + stripDate.ToString("MM", CultureInfo.InvariantCulture) + "/" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "|((Mon|Tue|Wed|Thu|Fri) )?(\\d{1,2}(st|nd|rd|th)? )?(" + stripDate.ToString("MMMM", CultureInfo.InvariantCulture) + "|" + stripDate.ToString("MMM", CultureInfo.InvariantCulture) + ")( \\d{1,2}(st|nd|rd|th)?| (" + stripDate.ToString("yy", CultureInfo.InvariantCulture) + "|" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "))?))?\\Z");
+            try
+            {
+                // Checks for match on ANY date in the string...
+                if (matchStripDate.IsMatch(name))
+                {
+                    // Date Found - now get the date value found
+                    string sDateFound = matchStripDate.Match(name).ToString();
+
+                    // Strip trouble characters eg ' and check for 'sept' and remove 't' also remove (st | nd | rd | th) and keep numeric
+                    sDateFound = sDateFound.Replace("'", string.Empty);
+                    sDateFound = sDateFound.Replace("Sept", "Sep").Replace("sept", "sep");
+                    sDateFound = removeDaySuffix.Replace(sDateFound, " ");
 
-            if (matchStripDate.IsMatch(name))
+                    // Convert to DateTime for comparison to similardate
+                    DateTime dtDateFound = Convert.ToDateTime(sDateFound, CultureInfo.CurrentCulture);
+
+                    // Check if date is within the +/- 'similardateday' range of 'date'
+                    if (dtDateFound >= stripDate.AddDays(-iSimilarDateDayRange) && dtDateFound <= stripDate.AddDays(iSimilarDateDayRange))
+                    {
+                       name = matchStripDate.Replace(name, string.Empty).ToString().Trim();
+                    }
+                }
+
+                return name;
+            }
+            catch
             {
-                name = matchStripDate.Match(name).Groups["name"].ToString();
+                return null;
             }
```

Comment:
If you remove this try...catch which swallows the exception and returns null that should cause the exception occurring during two of the tests to be revealed.